src/papi-client.ts

```diff
@@ -39,18 +53,21 @@ export class PapiClient {
         return this.promisify(this.requestInBatch(put, "/"));
     }
 
-    public waitForBatchSuccess(delay?: number): Promise<string> {
+    public waitForBatchSuccess(delay?: number): Promise<BatchResult> {
         this.assertBatchHasBeenStarted();
 
-        let deferred = Promise.defer<string>();
+        let deferred = Promise.defer<BatchResult>();
         let callback = (err, res) => {
             if (err) {
                 deferred.reject(err);
             } else if (res.body.status === "ERROR") {
                 deferred.reject("ERROR: " + res.body.operationDone + "/" + res.body.operationCount);
```

Comment:
The "reject" does not have to returns the same type than the "success". Rejecting with a string or an exception is a common practice I think.